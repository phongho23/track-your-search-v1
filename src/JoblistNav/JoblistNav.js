import React from 'react'
import { NavLink, Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import CircleButton from '../CircleButton/CircleButton'
import ApiContext from '../ApiContext'
import { countJobsForWeek } from '../jobs-helpers'
import './JoblistNav.css'

export default class JoblistNav extends React.Component {
  static contextType = ApiContext;
  
  render() {

    const { jobs=[], weeks=[] } = this.context

  return (
    <div className='JoblistNav'>
      <div className='JoblistNav__button-wrapper'>
        <CircleButton
          tag={Link}
          to='/home/add-week'
          type='button'
          className='JoblistNav__add-week-button'>
          <FontAwesomeIcon icon='plus' />
          <br />
          Week
        </CircleButton>
      </div>
      <ul className='JoblistNav__list'>
        {weeks.map(week =>
          <li style={{display: "inline-block"}} 
          key={week.id}
          >
            <NavLink
              className='JoblistNav__week-link'
              to={`/home/week/${week.id}`}
              >
              <span className='JoblistNav__num-jobs'>
                {countJobsForWeek(jobs, week.id)}
              </span>
              {week.name}
            </NavLink>
          </li>
        )}
      </ul>
    </div>
    )
  }
}