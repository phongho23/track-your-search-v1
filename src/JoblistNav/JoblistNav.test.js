import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import JoblistNav from './JoblistNav'

describe(`JoblistNav component`, () => {
  it('renders a .JoblistNav by default', () => {
    const wrapper = shallow(<JoblistNav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip('renders a link in ul for each week in array', () => {
    const context = {
      jobs: [
        {
            "id": "1",
            "name": "Software Engineer I", 
            "modified": "2021-01-03T00:00:00.000Z", 
            "weekId": "1", 
            "jobtitle": "Software Engineer III",
            "companyname": "Target",
            "postedurl": "http://www.URLholder.com/",
            "interview": "Scheduled for 05/20/2021",
            "jobrating": "5",
            "content": "Corporis accusamus placeat quas non voluptas. Harum fugit molestias qui. Velit ex animi reiciendis quasi."
        },
        {
            "id": "2",
            "name": "Software Engineer II", 
            "modified": "2021-08-15T23:00:00.000Z", 
            "weekId": "2", 
            "jobtitle": "Software Engineer I",
            "companyname": "Wal-Mart",
            "postedurl": "http://www.URLholder.com/",
            "interview": "Scheduled for 06/20/2021",
            "jobrating": "5",
            "content": "Eos laudantium quia ab blanditiis temporibus necessitatibus. Culpa et voluptas ut sed commodi."
        },
        {
            "id": "3",
            "name": "Software Engineer III", 
            "modified": "2021-03-01T00:00:00.000Z", 
            "weekId": "3", 
            "jobtitle": "Software Engineer II",
            "companyname": "Home Depot",
            "postedurl": "http://www.URLholder.com/",
            "interview": "Scheduled for 07/20/2021",
            "jobrating": "5",
            "content": "Occaecati dignissimos quam qui facere deserunt quia. Quaerat aut eos laudantium dolor odio officiis illum."
        }
      ],
      weeks: [
        {
          "id": "1",
          "name": "Week 1"
        },
        {
          "id": "2",
          "name": "Week 2"
        },
        {
          "id": "3",
          "name": "Week 3"
        }
      ]
    }
    const ul = shallow(<JoblistNav />, context)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
