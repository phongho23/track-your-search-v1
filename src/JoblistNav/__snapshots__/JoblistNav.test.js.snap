// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoblistNav component renders a .JoblistNav by default 1`] = `
<div
  className="JoblistNav"
>
  <div
    className="JoblistNav__button-wrapper"
  >
    <NavCircleButton
      className="JoblistNav__add-week-button"
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/home/add-week"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Week
    </NavCircleButton>
  </div>
  <ul
    className="JoblistNav__list"
  />
</div>
`;
